logging/logging.go

```diff
@@ -311,7 +311,12 @@ type templateEntryWriter struct {
 func (w templateEntryWriter) Write(p []byte) (n int, err error) {
 	e := *w.template
 	e.Payload = string(p)
-	w.l.Log(e)
+	// The second argument to logInternal() is how many frames to skip
+	// from the call stack when determining the source location. In the
+	// current implementation of log.Logger (i.e. Go's logging library)
```

Comment:
The "current implementation" part worries me a bit. Is there a way to find this value dynamically?

Although I it looks like the tests will warn us if this value changes in the future?

Reply:
Yes, this was my concern too. I'm not sure there's an easy way to determine the call depth correction dynamically. The Go logging library implementation also has such an assumption baked in, see https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/log/log.go;l=165-170;drc=c7942f87a2587ee989f6e282d887b4652119133a and https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/log/log.go;l=213;drc=c7942f87a2587ee989f6e282d887b4652119133a (as an example caller).

I don't think there's a way to avoid this. The only place we can inject this `SourceLocation` is in [`templateWriter.Write()`](https://github.com/googleapis/google-cloud-go/blob/c71b090f59317d43a2a5f2d9dd81c493b14d8908/logging/logging.go#L311). We have to make an assumption for call depth adjustment in any case.